MAINTAINING.md

```diff
@@ -49,6 +49,73 @@ Note `--distribution-type all`.
 
 ### New IDE version
 
+Each new major platform release brings not only a lot of new features
+but usually also a bunch of new incompatibilities (both source and binary ones).
+Since we want to support at least the latest stable major platform release and new [EAP](https://www.jetbrains.com/resources/eap/)s,
+we have to compile the plugin with different platform versions to avoid any runtime errors.
+
+To solve the problem with incompatibilities, we use conditional compilation based on Gradle [source sets](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSet.html).
+See [CONTRIBUTING.md](CONTRIBUTING.md#source-code-for-different-platform-versions) for more details about concrete source structure.
+
+#### How to support new platform version
+
+The following explanation uses `old`, `current` and `new` platform terms that mean:
+* `old` - number of the oldest supported platform version that should be dropped while
+* `current` - number of the latest major stable platform release
+* `new` - number of new platform that should be supported
+
+For example, at the moment of writing, `201` is `old`, `202` is `current` and `203` is `new`.
+ 
+Step by step instruction on how to support new platform version:
+* Drop all code related to the `old` platform version, i.e. drop `gradle-%old%.properties` and
+all `%module_name%/src/%old%` directories in each module
+* Move code from `%module_name%/src/%current%` directories into common source set,
+i.e. `%module_name%/src/%current%/main` into `%module_name%/src/main` and `%module_name%/src/%current%/test` into `%module_name%/src/test`.
+Also, simplify the moved code if possible.
+* Add support for `new` platform, i.e. add `gradle-%new%.properties` with all necessary properties and make it compile.
+It may be required to extract some code into platform-specific source sets to make plugin compile with each supported platform.
+See [Tips and tricks](#tips-and-tricks) section for the most common examples of how to do it
+* Fix tests
+* Update log path in run configurations like `runIDEA` and `runCLion`.
+For `runIDEA` configuration it requires:
+    - drop `idea-%old%.log` item 
+    - add `idea-%new%.log` item with `$PROJECT_DIR$/plugin/build/idea-sandbox-%new%/system/log/idea.log` path
+* Update CI workflows to use new platform version instead of old one.
+It usually requires just to update `platform-version` list in all workflows where we build the plugin.
+At the moment of writing they are `check.yml`, `rust-nightly.yml` and `rust-release.yml`
+* Fix `BACKCOMPAT: %old%` comments
+
+#### Tips and tricks
```

Comment:
Maybe add a note about the `IgnoreInPlatform` annotation and `ApplicationInfo.getInstance().build`?

Reply:
Added